src/Qir/Execution/Tests.Microsoft.Quantum.Qir.Tools/QirDriverGeneratorTests.cs

```diff
@@ -0,0 +1,184 @@
+                    new EntryPointOperation
+                    {
+                        Name = "UseResultArg",
+                        Parameters = new List<Parameter>{new Parameter{ Name = "ResultArg", Type = DataType.ResultType}}
+                    }
+                },
+                {
+                    "UseResultArrayArg",
+                    new EntryPointOperation
+                    {
+                        Name = "UseResultArrayArg",
+                        Parameters = new List<Parameter>{new Parameter{ Name = "ResultArrayArg", Type = DataType.ArrayType, ArrayType = DataType.ResultType}}
+                    }
+                },
+                {
+                    "UseStringArg",
+                    new EntryPointOperation
+                    {
+                        Name = "UseStringArg",
+                        Parameters = new List<Parameter>{new Parameter{ Name = "StringArg", Type = DataType.StringType}}
+                    }
+                },
+                {
+                    "UseMiscArgs",
+                    new EntryPointOperation
+                    {
+                        Name = "UseMiscArgs",
+                        Parameters = new List<Parameter>{
+                            new Parameter{ Name = "BoolArg", Type = DataType.BoolType},
+                            new Parameter{ Name = "IntegerArrayArg", Position = 1, Type = DataType.ArrayType, ArrayType = DataType.IntegerType},
+                            new Parameter{ Name = "RangeArg", Position = 2, Type = DataType.RangeType},
+                            new Parameter{ Name = "StringArg", Position = 3, Type = DataType.StringType}
+                        }
+                    }
+                }
+            };
+
+        private static string RemoveLineEndings(string str) =>
+            str.Replace("\r\n", string.Empty).Replace("\n", string.Empty).Replace("\r", string.Empty);
+
+        [Theory]
+        [InlineData("UseNoArgs")]
+        [InlineData("UseBoolArg")]
+        [InlineData("UseBoolArrayArg")]
+        [InlineData("UseDoubleArg")]
+        [InlineData("UseDoubleArrayArg")]
+        [InlineData("UseIntegerArg")]
+        [InlineData("UseIntegerArrayArg")]
+        [InlineData("UsePauliArg")]
+        [InlineData("UsePauliArrayArg")]
+        [InlineData("UseRangeArg")]
+        [InlineData("UseRangeArrayArg")]
+        [InlineData("UseResultArg")]
+        [InlineData("UseResultArrayArg")]
+        [InlineData("UseStringArg")]
+        [InlineData("UseMiscArgs")]
+        public void GenerateFullStateSimulatorDriver(string testCase)
+        {
+            var entryPointOperation = TestCases[testCase];
+            var driverGenerator = new QirFullStateDriverGenerator();
+            var driverFileName = $"{testCase}.cpp";
+            var verificationCppSourceCode = RemoveLineEndings(File.ReadAllText(Path.Combine(TestCasesDirectory, driverFileName)));
+            if (!Directory.Exists(TestArtifactsDirectory))
```

Comment:
CreateDirectory does nothing if directory already exists. This conditional is not needed.

Reply:
Removed conditional.